### PR TITLE
add ppc_*_overlay_grouped functions

### PR DESCRIPTION
**Pull request by tjmahr**
``` r
library(bayesplot)
#> This is bayesplot version 1.7.1
#> - Online documentation and vignettes at mc-stan.org/bayesplot
#> - bayesplot theme set to bayesplot::theme_default()
#>    * Does _not_ affect other ggplot2 plots
#>    * See ?bayesplot_theme_set for details on theme setting
y <- example_y_data()
yrep <- example_yrep_draws()
group <- example_group_data()

ppc_dens_overlay_grouped(y, yrep[1:20,], group = group)
```

``` r
ppc_ecdf_overlay_grouped(y, yrep[1:20,], group = group)
```

<sup>Created on 2019-12-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by tjmahr** [comment]
The plots now have axis lines. (Note the addition of the vertical y-axis lines compared to the original versions above.)

``` r
library(bayesplot)
#> This is bayesplot version 1.7.1
#> - Online documentation and vignettes at mc-stan.org/bayesplot
#> - bayesplot theme set to bayesplot::theme_default()
#>    * Does _not_ affect other ggplot2 plots
#>    * See ?bayesplot_theme_set for details on theme setting
y <- example_y_data()
yrep <- example_yrep_draws()
group <- example_group_data()

ppc_dens_overlay_grouped(y, yrep[1:20,], group = group)
```



``` r
ppc_ecdf_overlay_grouped(y, yrep[1:20,], group = group)
```



<sup>Created on 2019-12-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by tjmahr** [comment]
Ready for review.

Travis won't build it because of BH issues and rhub fails when trying to install BiocManager but the checks on my Windows 10 machine all pass. 

I do get warnings during tests about our use of `expand_scale()`, but that's because I have the dev version of ggplot2 where `expand_scale()` has been deprecated.

**Comment by tjmahr** [comment]
Oops, this breaks code from the brms package.

```
pp_check(model)
 Error: Argument 'group' is required for ppc type 'dens_overlay'. 
```

**Comment by jgabry** [comment]
Wow, sorry it took me _forever_ to get to this @tjmahr! This looks great. 

> Oops, this breaks code from the brms package.

@paul-buerkner is there something in brms that can be changed for this to not cause an error?

**Comment by jgabry** [comment]
@tjmahr Coming back to this, I looked into the brms issue and I think the reason this causes an error in brms is that as part of this PR `ppc_dens_overlay()` gains a group argument too (instead of just the `ppc_dens_overlay_grouped()` function having the `group` argument. brms thinks any functions with the `group` argument require the group argument (which has always been the case).

**Comment by jgabry** [comment]
I think if we wait on this until we deal with #151 then we can use the new mechanism for the grouped plots calling the ungrouped version. That should get around the brms error.

**Comment by jgabry** [comment]
> pp_check(model)
>  Error: Argument 'group' is required for ppc type 'dens_overlay'.

#253 which was just opened reminded me of this. @tjmahr is there a reason `ppc_dens_overlay` needs to gain a `group` argument? Can that argument exclusively be in `ppc_dens_overlay_grouped`? That would get around the brms error that was preventing us from merging this. Otherwise we might need to wait for the big PR https://github.com/stan-dev/bayesplot/pull/222 before merging this.

**Comment by tjmahr** [comment]
``` r
library(bayesplot)
#> This is bayesplot version 1.7.2.9000
#> - Online documentation and vignettes at mc-stan.org/bayesplot
#> - bayesplot theme set to bayesplot::theme_default()
#>    * Does _not_ affect other ggplot2 plots
#>    * See ?bayesplot_theme_set for details on theme setting

y <- example_y_data()
yrep <- example_yrep_draws()
group <- example_group_data()

ppc_dens_overlay_grouped(y, yrep[1:20,], group = group)
```



``` r
y <- example_y_data()
yrep <- example_yrep_draws()
group <- example_group_data()

ppc_ecdf_overlay_grouped(y, yrep[1:20,], group = group)
```



<sup>Created on 2020-12-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by tjmahr** [comment]
> 
> 
> > pp_check(model)
> > Error: Argument 'group' is required for ppc type 'dens_overlay'.
> 
> #253 which was just opened reminded me of this. @tjmahr is there a reason `ppc_dens_overlay` needs to gain a `group` argument? Can that argument exclusively be in `ppc_dens_overlay_grouped`? That would get around the brms error that was preventing us from merging this. Otherwise we might need to wait for the big PR #222 before merging this.

I removed the group argument from `ppc_dens_overlay()` and used a ggplot2 idiom to keep from writing the same plotting code twice.

File: R/ppc-distributions.R [hunk 1]
```diff
@@ -357,6 +355,11 @@ ppc_dens_overlay_grouped <- function(
     kernel = kernel,
     n_dens = n_dens
   )
+  # Use + list(data) trick to replace the data in the plot. The layer-specific
+  # data in the y and yrep layers should be safe because they are
+  # specified using a function on the main plot data.
+  data <- ppc_data(y, yrep, group = group)
+  p_overlay <- p_overlay + list(data)
```
Comment (jgabry) [inline R/ppc-distributions.R:362]: I didn't even know about this trick!

Comment (jgabry) [reply R/ppc-distributions.R:362]: Or I maybe did but had since forgotten. Either way, this seems seems good.

**Review by jgabry — APPROVED** [review]
Thanks for updating this @tjmahr. I think it's ready to go but will let all the checks finish running.

**Pull request merged**